### PR TITLE
Add some commonly used QL snippets

### PR DESCRIPTION
**Pull request by edoardopirovano**
This PR addresses #392 by introducing some simple snippets for query metadata, classes, from/where/select statements, predicates, dataflow/taint tracking classes, and all the aggregates and quantifiers.

### CODE REVIEW

File: extensions/ql-vscode/snippets.json [hunk 1]
```diff
@@ -0,0 +1,134 @@
+{
+  "Query Metadata": {
+    "prefix": "querymetadata",
+    "body": [
+      "/**",
+      " * @name $1",
+      " * @description $2",
+      " * @kind $3",
```
Comment (aeisenberg) [inline extensions/ql-vscode/snippets.json:8]: Either add a _choice_ here, or separate this into multiple snippets, one for each metadata kind.

https://code.visualstudio.com/docs/editor/userdefinedsnippets#_choice

File: extensions/ql-vscode/snippets.json [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    "description": "Metadata for a query"
+  },
+  "Class": {
+    "prefix": "class",
+    "body": ["class $1 extends $2 {", "\t$0", "}"],
+    "description": "A class"
+  },
+  "From/Where/Select": {
+    "prefix": "from",
+    "body": ["from $1", "where $2", "select $3"],
```
Comment (aeisenberg) [inline extensions/ql-vscode/snippets.json:22]: I find it easier to read if there are default values for each replacement step.

```suggestion
    "body": ["from ${1:params}", "where ${2:clause}", "select ${2:args}"],
```

**Review by aeisenberg — CHANGES_REQUESTED** [review]
This looks nicely comprehensive. Thanks!

I think in general, any snippet placeholder that _must_ be added should have a default value. If nothing else, this just signifies where a user can edit.

**Review by aeisenberg — APPROVED** [review]
Nice work!

**Comment by aeisenberg** [comment]
@github/docs-content-dsp Is this the kind of thing that should go in the documentation?

**Comment by shati-patel** [comment]
> @github/docs-content-dsp Is this the kind of thing that should go in the documentation?

Looks exciting 😄 I've opened an internal docs issue where we can assess if/where we need to document this.

**Pull request merged**